### PR TITLE
Create the initial app component hierarchy, loading it in the index.html entry point

### PR DESCRIPTION
**Pull request by trusktr**
Note, `index.html` is the only HTML file we need. The rest of the pages are loaded dynamically within the logic of the `MainContent` class (the `<main-content>` element).

To run it, the following will build then open a browser tab with the app:

```
npm i
npm start
```

To understand the changes,

- start in `index.html` (note that the other `.html` files were deleted).
- `index.html` contains only the stuff needed to bootstrap the app (links to CSS files, link to `index.js`, etc).
- `index.js` is the JS-side entry point, which loads (the output wasm module of) `assembly/index.ts` which is the AS-side entry point.
- The AS-side entry point creates the top-level `<aspkg-app>` element and sticks into the `<body>`, then makes the loading animation fade away.
- The `<aspkg-app>` element (`AspkgApp.ts`) sets up the common layout for all pages including `<nav-bar>`, `<main-content>`, and `<the-footer>`. This encapsulates the top-level structure of the app.
- The content of `<nav-bar>` (`NavBar.ts`) and `<the-footer>` (`TheFooter.ts`, notice the pattern in naming of elements and their classes and files) is for now static and doesn't change.
- The `<main-content>` element (`MainContent.ts`) how the initial workings of a mechanism that changes which page we will render. Currently it emulates route changes with a `setInterval` to rotate between the initial route definitions (we will expand on this) every 2 seconds. Every two seconds the main content area of the app will change to render the specified element defined in the particular route that we are currently at.
- Once we have an actual routing implementation in place, the URL in the address bar will actually dictate which page to render, and `MainContent`'s routing will operate based on whatever the current URL in the address bar is.
- Note that soon, when the address bar URL will be changing due to user actions, the whole page will not to a full refresh, instead it will tell our app which route element to render and only the main content area changes as you currently see with the `setInterval` emulation.

I don't know if you know about this type of dynamic routing. If you want to see an example, the LUME docs site is one. When you change pages, it doesn't do a full refresh, just swaps the content of the main content area: https://lume.io/docs/#/guide/positioning

It is made with this project I am helping to maintain: http://docsify.js.org/

### CODE REVIEW

File: assembly/elements/pages/PackageDetails.ts [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+							font-weight: normal;
+							padding-left: 15px;
+							width: auto;
+							margin-bottom: 16px;
+						"
+						href="https://github.com/aspkg/example"
+					>
+						-
+					</p>
+					<p
```
Comment (trusktr) [inline assembly/elements/pages/PackageDetails.ts:81]: This markup is from BSS output, pretty messy with all the duplicated inline styles. I haven't cleaned this one up yet.

**Comment by trusktr** [comment]
You'll see some `/*html*/` comments in the code. This tells my VS Code to highlight the content of the string as HTML syntax. I'm using this nifty extension: https://marketplace.visualstudio.com/items?itemName=bierner.comment-tagged-templates

I also have this one: https://marketplace.visualstudio.com/items?itemName=Tobermory.es6-string-html
Not sure which plugin is actually handling the `/*html*/` comments specifically.

The first plugin allows many other syntaxes too. F.e. `/*js*/` will highlight JS inside template strings.

File: assembly/elements/pages/PackageDetails.ts (hunk 1 above)
Comment (JairusSW) [reply assembly/elements/pages/PackageDetails.ts:81]: Yeah, that's what happens for drag-drop builders. haha!

**Comment by JairusSW** [comment]
Looks good! Thanks!

**Pull request merged**